pkg/ddl/job_scheduler.go

```diff
@@ -692,45 +692,77 @@ func insertDDLJobs2Table(ctx context.Context, se *sess.Session, jobWs ...*JobWra
 	return errors.Trace(err)
 }
 
-func job2SchemaIDs(job *model.Job) string {
-	return job2UniqueIDs(job, true)
-}
+func makeStringForIDs(ids []int64) string {
+	set := make(map[int64]struct{}, len(ids))
+	for _, id := range ids {
+		set[id] = struct{}{}
+	}
 
-func job2TableIDs(job *model.Job) string {
-	return job2UniqueIDs(job, false)
+	s := make([]string, 0, len(set))
+	for id := range set {
+		s = append(s, strconv.FormatInt(id, 10))
+	}
+	slices.Sort(s)
+	return strings.Join(s, ",")
 }
 
-func job2UniqueIDs(job *model.Job, schema bool) string {
+func job2SchemaIDs(job *model.Job) string {
 	switch job.Type {
-	case model.ActionExchangeTablePartition, model.ActionRenameTables, model.ActionRenameTable:
+	case model.ActionRenameTables:
 		var ids []int64
-		if schema {
-			ids = job.CtxVars[0].([]int64)
+		if job.Version == model.JobVersion1 {
+			// After json.Marshal and json.Unmarshall, the type is changed to pointer
+			if _, ok := job.Args[0].([]int64); ok {
+				ids = append(job.Args[0].([]int64), job.Args[1].([]int64)...)
+			} else {
+				ids = append(*job.Args[0].(*[]int64), *job.Args[1].(*[]int64)...)
+			}
 		} else {
-			ids = job.CtxVars[1].([]int64)
-		}
-		set := make(map[int64]struct{}, len(ids))
-		for _, id := range ids {
-			set[id] = struct{}{}
+			arg := job.Args[0].(*model.RenameTablesArgs)
```

Comment:
you can pass args in JobWrapper now, see https://github.com/pingcap/tidb/blob/b1767ca12464c45f741ace20b9ec679d2ea4de5f/pkg/ddl/job_scheduler.go#L732